src/Arcus.Testing.Integration.DataFactory/TemporaryDataFlowDebugSession.cs

```diff
@@ -405,6 +412,42 @@ public RunDataFlowOptions AddDataFlowParameter(string name, object value)
             return this;
         }
 
+        /// <summary>
+        /// Adds a parameter to a DataSet present the DataFlow to run.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="datasetName"/> is blank.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="name"/> is blank.</exception>
+        public RunDataFlowOptions AddDataSetParameter(string datasetName, string name, object value)
+        {
+            if (string.IsNullOrWhiteSpace(datasetName))
+            {
+                throw new ArgumentException("DataSet name should not be blank", nameof(datasetName));
+            }
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException("DataSet parameter name should not be blank", nameof(name));
+            }
+
+            IDictionary<string, object> dataSetParameters;
+            if (DataSetParameters.TryGetValue(datasetName, out BinaryData keyValue))
+            {
+                dataSetParameters = JsonSerializer.Deserialize<IDictionary<string, object>>(keyValue);
+                dataSetParameters.Add(name, value);
+            }
+            else
+            {
+                dataSetParameters = new Dictionary<string, object>
+                {
+                    { name, value }
+                };
+            }
+
+            string jsonString = JsonSerializer.Serialize(dataSetParameters);
```

Comment:
Can we change this so that we only JSON seralize upon running the dataflow instead of doing serialization/deserialization on each added parameter? The `internal` exposed parameters can be that 'dictionary in dictionary' one and adding parameters here will become a lot easier with the benefit of better performance.